### PR TITLE
Add freebsd 14

### PR DESCRIPTION
**Pull request by am11**

### CODE REVIEW

File: src/ubuntu/build-scripts/run-arcade-build-rootfs.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ fi
 echo "Using $dockerCrossDepsTag to clone arcade to fetch scripts used to build cross-toolset"
 docker exec $buildRootFSContainer sh -c '
     git config --global user.email builder@dotnet-buildtools-prereqs-docker &&
-    git clone --depth 1 --single-branch https://github.com/dotnet/arcade /scripts'
+    git clone --depth 1 --single-branch https://github.com/am11/arcade -b patch-7 /scripts'
```
Comment (am11) [inline src/ubuntu/build-scripts/run-arcade-build-rootfs.sh:47]:
```suggestion
    git clone --depth 1 --single-branch https://github.com/dotnet/arcade /scripts'
```

**Comment by am11** [comment]
cc @wfurt, @akoeplinger, @Thefrank
freeebsd14 was picked up, from logs:

```
2024-01-18T16:48:03.9022275Z #7 [builder 2/2] RUN /scripts/eng/common/cross/build-rootfs.sh freebsd14 x64
2024-01-18T16:48:04.2025619Z #7 0.310 I: Retrieving InRelease 
2024-01-18T16:48:05.4058347Z #7 1.439 I: Checking Release signature
2024-01-18T16:48:05.4058732Z #7 1.447 I: Valid Release signature (key id 790BC7277767219C42C86F933B4FE6ACC0B21F32)
2024-01-18T16:48:06.3073204Z #7 2.328 I: Retrieving Packages 
2024-01-18T16:48:07.4940872Z #7 3.664 I: Validating Packages 
2024-01-18T16:48:07.7754211Z #7 3.838 I: Resolving dependencies of required packages...
2024-01-18T16:48:07.7754463Z #7 3.945 I: Resolving dependencies of base packages...
2024-01-18T16:48:08.6779781Z #7 4.734 I: Checking component main on http://archive.ubuntu.com/ubuntu...
2024-01-18T16:48:08.8285594Z #7 4.914 I: Retrieving adduser 3.113+nmu3ubuntu4
2024-01-18T16:48:09.7310879Z #7 5.827 I: Validating adduser 3.113+nmu3ubuntu4
2024-01-18T16:48:09.7311181Z #7 5.845 I: Retrieving apt 1.2.10ubuntu1
2024-01-18T16:48:10.6327404Z #7 6.691 I: Validating apt 1.2.10ubuntu1
...
```

Please merge https://github.com/dotnet/arcade/pull/14387 first then commit the suggestion above (https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/961#discussion_r1457762173) and merge this one. Thanks!

**Comment by am11** [comment]
@Thefrank, I'm sorry I didn't realized you already had that PR. 🙈

**Review by wfurt — APPROVED** [review]
LGTM

**Pull request merged**